Subject: Linux-RT 2.6.27-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Tony Jones <tonyj@suse.de>
Subject: NOHZ: local_softirq_pending with tickless

From: Mikulas Patocka <mikulas@artax.karlin.mff.cuni.cz>

On one of my machines with tickless kernel and plip I get messages :

NOHZ: local_softirq_pending 08

always when using plip (on other machine with tickless kernel and plip I 
get no errors). Thebug happens both on 2.6.21 and 2.6.22-rc1

This patch fixes that. Note that plip calls netif_rx neither from hardware 
interrupt nor from ksoftirqd, so there is no one who would wake 
ksoftirqd then. netif_tx calls only 
__raise_softirq_irqoff(NET_RX_SOFTIRQ), which sets softirq bit, but 
doesn't wake ksoftirqd.

Mikulas

Signed-off-by: Mikulas Patocka <mikulas@artax.karlin.mff.cuni.cz>

Removed the remaining users of __raise_softirq_irqoff() as well.

	tglx

---
 net/core/dev.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/core/dev.c b/net/core/dev.c
index 194e4d1..877d44a 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -2452,7 +2452,7 @@ out:
 
 softnet_break:
 	__get_cpu_var(netdev_rx_stat).time_squeeze++;
-	__raise_softirq_irqoff(NET_RX_SOFTIRQ);
+	raise_softirq_irqoff(NET_RX_SOFTIRQ);
 	goto out;
 }
 
